data/quot.lean

```diff
@@ -7,6 +7,34 @@ Quotients -- extends the core library
 -/
 variables {α : Sort*} {β : Sort*}
 
+namespace quot
+variables {ra : α → α → Prop} {rb : β → β → Prop} {φ : quot ra → quot rb → Sort*}
+local notation `⟦`:max a `⟧` := quot.mk _ a
+
+protected def hrec_on₂ (qa : quot ra) (qb : quot rb)
+  (refla : reflexive ra) (reflb : reflexive rb) (f : ∀ a b, φ ⟦a⟧ ⟦b⟧)
+  (c : ∀ a₁ b₁ a₂ b₂, ra a₁ a₂ → rb b₁ b₂ → f a₁ b₁ == f a₂ b₂) : φ qa qb :=
```

Comment:
You should not need reflexivity for this. Did you try iterated `hrec_on`?

Reply:
No, I'm not sure what you mean.

Reply:
Oh, I see. You only need the assumption given the form of the equality proof. How about stating it like this:
```
protected def hrec_on₂ (qa : quot ra) (qb : quot rb) (f : ∀ a b, φ ⟦a⟧ ⟦b⟧)
  (ca : ∀ {b a₁ a₂}, ra a₁ a₂ → f a₁ b == f a₂ b)
  (cb : ∀ {a b₁ b₂}, rb b₁ b₂ → f a b₁ == f a b₂) : φ qa qb :=
```
You can keep the one-hyp version for `quotient`.

Reply:
Good idea. Done.